### ISSUE TITLE
Performance of this model

### ISSUE THREAD

**Issue by lv2020** [denizetkar/TreeGAN#1]
Hi, thanks for you share. I'm doing a similar research on code generation and I also tried this model. But I haven't got a useful result yet. What I find is that generator only works on first 50 timesteps, and then it starts to generate some meaningless codes. So could you please share some details about this model, for example, the generated results, training settings and parameters? I think it is the problem of policy loss. Even in Seq GAN, they used sequences of length 20 to train the model, so it may be hard for the Tree GAN to work on a longer sequence. In my case, the median and mean length of training data are 837 and 900 respectively. Thanks.

**Comment by denizetkar** [comment]
Thank you for the kind words and your interest. I must admit that this project is not complete yet. Because, it is still missing the training loop as well as the hyperparameter optimization. I'm currently doing a master's degree and I will hopefully finalize this project as soon as I get some free time.
I also should have mentioned in the README file that I was inspired by the following paper: https://arxiv.org/abs/1808.07582

**Comment by lv2020** [comment]
Thanks for your quick response. Actually I'm inspired by the same paper and this repo is the only related one I found in GitHub. Still, I doubt whether it works because C language has a much longer and deeper AST than examples in papers. Anyway, I think I will try some things like reduce the grammar rules to train this model.
